api/app/auto_spatial_advisory/process_fuel_type_area.py

```diff
@@ -112,11 +111,63 @@ def intersect_raster_by_advisory_shape(threshold: int, advisory_shape_id: int, s
     :param temp_dir: A temporary location for storing intermediate files.
     """
     output_path = os.path.join(temp_dir, f"advisory_shape_{source_identifier}_threshold_{threshold}.tif")
-    warp_options = gdal.WarpOptions(format="GTiff", cutlineDSName=DB_READ_STRING, cutlineSQL=f"SELECT geom FROM advisory_shapes WHERE id={advisory_shape_id}", cropToCutline=True)
+
+    raster = gdal.Open(raster_path)
+    tif_prj = raster.GetProjection()
+    tif_srs = osr.SpatialReference(wkt=tif_prj)
+    raster = None
+
+    advisory_shape = get_advisory_shape(advisory_shape_id, temp_dir, tif_srs)
+    warp_options = gdal.WarpOptions(format="GTiff", cutlineDSName=advisory_shape, cropToCutline=True)
     gdal.Warp(output_path, raster_path, options=warp_options)
     return output_path
 
 
+def get_advisory_shape(advisory_shape_id: int, out_dir: str, projection: osr.SpatialReference) -> str:
+    """
+    Get advisory_shape from database and store it (typically temporarily) in the specified projection for raster
+    intersection. The advisory_shape layer returned by ExecuteSQL must be stored somewhere and can't simply be returned
+    because of the way GDAL connection references are handled (https://gdal.org/api/python_gotchas.html)
+
+    :param advisory_shape_id: advisory_shape_id
+    :type advisory_shape_id: int
+    :param out_dir: Output directory of reprojected polygon(s)
+    :type out_dir: str
+    :param projection: Spatial reference
+    :type projection: osr.SpatialReference
+    :return: path to stored advisory shape
+    :rtype: str
+    """
+    data_source = ogr.Open(DB_READ_STRING)
+    sql = f'SELECT geom FROM advisory_shapes WHERE id={advisory_shape_id}'
+    advisory_layer = data_source.ExecuteSQL(sql)
+
+    advisory_shape = reproject_ogr_layer(advisory_layer, out_dir, projection)
+
+    data_source = None
+
+    return advisory_shape
+
+
+def reproject_ogr_layer(layer: ogr.Layer, out_dir: str, to_projection: osr.SpatialReference) -> str:
+    layer_srs = layer.GetSpatialRef()
+    coordinate_transform = osr.CoordinateTransformation(layer_srs, to_projection)
+
+    output_path = f'/{out_dir}/reprojected.gpkg'
+    reprojected_ds = ogr.GetDriverByName('GPKG').CreateDataSource(output_path)
+    reprojected_layer = reprojected_ds.CreateLayer('reprojected', srs=to_projection)
+
+    for feature in layer:
+        geometry = feature.GetGeometryRef()
+        geometry.Transform(coordinate_transform)
+        new_feature = ogr.Feature(reprojected_layer.GetLayerDefn())
+        new_feature.SetGeometry(geometry)
+        reprojected_layer.CreateFeature(new_feature)
+        new_feature = None
+
+    return output_path
```

Comment:
Should we be setting reprojected_ds to None before returning? I've read mixed reviews on allowing garbage collection to handle data source clean up versus explicitly closing the references.

Reply:
Also not sure if layers should be set to None or not?!

Reply:
Good questions. I've read once the objects are out of scope the garbage collector should [handle it](https://gdal.org/api/python_gotchas.html#:~:text=This%20is%20typically%20unnecessary%20because%20these%20objects%20are%20automatically%20destroyed%20during%20garbage%20collection%20when%20no%20references%20to%20the%20Python%20object%20remain.). But it also seems it's a good idea to clean up datasource objects explicitly according to GDAL at the link above.

On another note, I saw while doing this that in GDAL 3.8 context managers are available.
`with ogr.Open() as.....`

Reply:
I believe our base image is running a super old GDAL version (3.3 or 3.4?) so we should consider rebuilding it soon.